merkle-patricia-tree/src/Data/Tree/MerklePatricia.hs

```diff
@@ -82,23 +85,62 @@ import qualified Data.ByteString.Char8 as B8
 
 -- * Type
 
-type Prefix = String
-
 -- | An opaque 'MerklePatriciaTree', parameterized by a hash algorithm and a
 -- type for its values.
 data MerklePatriciaTree alg a
-  = Leaf Prefix (Digest alg) a
-  | Node Prefix (Digest alg) [(Char, MerklePatriciaTree alg a)]
+  = Leaf [Bit] (Digest alg) a
+  | Node [Bit] (Digest alg) [(Bit, MerklePatriciaTree alg a)]
   deriving (Show)
 
 instance HashAlgorithm alg => Eq (MerklePatriciaTree alg a) where
   (root -> a) == (root -> b) = a == b
 
 -- | An opaque 'Proof' which can be used to prove membership or atomic
 -- operations such as 'add' or 'delete'
-newtype Proof alg = Proof {unProof :: [(Prefix, [(Char, Digest alg)])]}
+newtype Proof alg = Proof {unProof :: [([Bit], [(Bit, Digest alg)])]}
   deriving stock (Show)
 
+data Bit = Up | Down deriving (Show, Eq, Ord)
```

Comment:
IIRC its rather called `High | Low`.. coming from the voltage levels used to indicate signals